packages/tofu/src/referenceTracker.js

```diff
@@ -162,55 +270,98 @@ const usageDetectors = {
       }
     }
   },
+  /**
+   * @type {Detector<import('@babel/types').CallExpression>}
+   */
   CallExpression: (targetPath) => {
     // usages are where the result goes, and the containing function
     const fnResultUsages = tracePathToUsages(targetPath.parentPath)
     return fnResultUsages
   },
+  /**
+   * @type {Detector<import('@babel/types').ReturnStatement>}
+   */
   ReturnStatement: (targetPath) => {
     // usages are all references to this function
     const fnUsages = traceFunctionDeclarationToUsages(targetPath.scope.path)
     return fnUsages
   },
+  /**
+   * @type {Detector<import('@babel/types').VariableDeclarator>}
+   */
   VariableDeclarator: (targetPath) => {
     // value is being stored in a reference, track usage
-    const variableName = targetPath.node.id.name
-    const variableBinding = targetPath.scope.getBinding(variableName)
-    const bindingUsages = makeReferenceLinkagesFromBinding(variableBinding)
-    return bindingUsages
+    if ('name' in targetPath.node.id) {
+      const variableName = targetPath.node.id.name
+      const variableBinding = targetPath.scope.getBinding(variableName)
+      const bindingUsages = makeReferenceLinkagesFromBinding(variableBinding)
+      return bindingUsages
+    }
+    return []
   },
+  /**
+   * @type {Detector<import('@babel/types').MemberExpression>}
+   */
   MemberExpression: (targetPath) => {
     const leftSide = targetPath.node.object
     if (leftSide.type === 'Identifier') {
       const binding = targetPath.scope.getBinding(leftSide.name)
       if (binding) {
-        return new ReferenceLinkage({
-          path: binding.path,
-          label: 'parent of member',
-        })
+        return [
```

Comment:
why is this an array now?

Reply:
because a `Detector`--defined as per the usage--expects an array.  the result is returned by [`tracePathTousages()`](https://github.com/LavaMoat/LavaMoat/pull/761/files/b09ea4b4854b71b820c3ed201d174ccf105d31fa#diff-45f2498bbc6fc41108f02cdfc08366bdec96fb18e677cb875b6acdfede5cdc0fR343) and [`walkUsage()`](https://github.com/LavaMoat/LavaMoat/pull/761/files/b09ea4b4854b71b820c3ed201d174ccf105d31fa#diff-45f2498bbc6fc41108f02cdfc08366bdec96fb18e677cb875b6acdfede5cdc0fR108) iterates over it.

Looks like a bug to me, but I could be missing something.

Reply:
I'd defer merging until we know for sure. I was expecting a more "you missed that obvious thing there" answer.